File: library/book.cpp
#include "book.h"

Book::Book(vector <string> data) :
	Media(data[0], data[1], data[2]),
	m_call_num(data[0]),
	m_title(data[1]),
	m_subjects(data[2]),
	m_author(data[3]),
	m_description(data[4]),
	m_publisher(data[5]),
	m_city(data[6]),
	m_year(data[7]),
	m_series(data[8]),
	m_notes(data[9])
{}

bool Book::compare_other(const string term)
{
	if (m_description.find(term) != string::npos ||
		m_notes.find(term) != string::npos ||
		m_year.find(term) != string::npos)
		return true;
	else
		return false;
}

void Book::display() const
{
	cout << "--------------------------\n";
	cout << "Call Number: " << m_call_num << endl;
	cout << "Title: " << m_title << endl;
	cout << "Subjects: " << m_subjects << endl;
	cout << "Author: " << m_author << endl;
	cout << "Description: " << m_description << endl;
	cout << "Publisher: " << m_publisher << endl;
	cout << "City: " << m_city << endl;
	cout << "Year: " << m_year << endl;
	cout << "Series: " << m_series << endl;
	cout << "Notes: " << m_notes << endl;
}

File: final-exam/01/book.py
from media import Media


class Book(Media):

    description = ''
    year = ''
    author = ''
    publisher = ''
    series = ''
    city = ''

    def __init__(self, callNumber, title, subject, author, notes, description, publisher, city, series, year):
        Media.__init__(self, callNumber, title, subject, notes)
        self.description = description
        self.year = year
        self.author = author
        self.publisher = publisher
        self.series = series
        self.city = city
        self.items.append('Description: ' + description)
        self.items.append('Year: ' + year)
        self.items.append('Author: ' + author)
        self.items.append('Publisher: ' + publisher)
        self.items.append('City: ' + city)
        self.items.append('Series: ' + series)
        self.items.insert(0, 'Book:')

    def otherContains(self, word):
        return self.notes.find(word) != -1 or self.description.find(word) != -1 or self.year.find(word) != -1
